R/run_prep_calculation_loans.R

```diff
@@ -326,6 +331,92 @@ run_prep_calculation_loans <- function(input_path_project_specific,
+      by = c("company_name" = "name_company")
+    ) %>%
+    dplyr::select(-.data$id) %>%
+    dplyr::rename(id = .data$company_id)
+
+  # This aggregates AR PAMS to company_technology level to get emission factors
+  # Note that this is done including all assets, i.e. we look at global
+  # emission factors here
+  # TODO: run for each scenario geography
+  ar_pams <- ar_pams %>%
+    dplyr::select(-dplyr::starts_with("Direct")) %>%
+    tidyr::pivot_longer(
+      cols = dplyr::starts_with("Total"),
+      names_to = "year",
+      names_prefix = "Total ",
+      values_to = "ald_production"
+    ) %>%
+    dplyr::rename(
+      id = .data$`Company ID`,
+      company_name = .data$`Company Name`,
+      ald_sector = .data$`Asset Sector`,
+      technology = .data$`Asset Technology`,
+      technology_type = .data$`Asset Technology Type`,
+      ald_location = .data$`Asset Country`,
+      emissions_factor = .data$`Emissions Factor`,
+      emissions_factor_unit = .data$`Emissions Factor Unit`,
+      ald_production_unit = .data$`Activity Unit`
+    ) %>%
+    dplyr::mutate(
+      technology = dplyr::case_when(
+        .data$ald_sector == "Coal" ~ "Coal",
+        .data$technology %in% c("Gas", "Natural Gas Liquids") ~ "Gas",
+        .data$technology == "Oil and Condensate" ~ "Oil",
+        TRUE ~ .data$technology
+      )
+    ) %>%
+    dplyr::group_by(
+      .data$company_name, .data$id , .data$ald_sector, .data$ald_location,
+      .data$technology, .data$year, .data$ald_production_unit,
+      .data$emissions_factor, .data$emissions_factor_unit
+    ) %>%
+    dplyr::summarise(
+      ald_production = sum(.data$ald_production, na.rm = TRUE),
+      .groups = "drop"
+    ) %>%
+    dplyr::ungroup() %>%
+    dplyr::transmute(
+      id = as.numeric(id),
+      company_name = tolower(as.character(.data$company_name)),
+      ald_sector = as.character(.data$ald_sector),
+      ald_location = as.character(.data$ald_location),
+      technology = as.character(.data$technology),
+      year = as.numeric(.data$year),
+      ald_production = as.numeric(.data$ald_production),
+      ald_production = dplyr::if_else(.data$ald_production <= 0, 0, .data$ald_production),
+      ald_production_unit = as.character(.data$ald_production_unit),
+      ald_emissions_factor = as.numeric(.data$emissions_factor),
+      ald_emissions_factor_unit = as.character(.data$emissions_factor_unit)
+    ) %>%
+    dplyr::group_by(
+      .data$id, .data$company_name, .data$ald_sector, .data$technology, .data$year
+    ) %>%
+    dplyr::summarise(
+      # TODO: might introduce NaNs... check what to do here
+      plan_emission_factor = stats::weighted.mean(
+        x = .data$ald_emissions_factor, w = .data$ald_production, na.rm = TRUE
+      ),
+      plan_tech_prod = sum(.data$ald_production, na.rm = TRUE),
+      .groups = "drop"
+    ) %>%
+    dplyr::ungroup()
+
+  # Join EFs based on PAMS
+  # FIXME: we are losing rows here, which we should not
```

Comment:
any more insights on this?

Reply:
this was because of low carbon technologies that have zero buildout. They will not be in the asset based data, which is used to add EFs. These technologies need to remain in the loan book though, because the SMSP may require a buildout in such technologies, even if no such plans exist yet. I solved it with a left_join and subsequent imputation of zero value EFs for these cases.